.changeset/three-poems-beg.md

```diff
@@ -0,0 +1,23 @@
+---
+"effect": minor
+---
+
+use LazyArg for Effect.if branches
```

Comment:
Just out of curiosity, what's the rationale behind this change?

Reply:
See https://github.com/Effect-TS/effect/issues/2345